testenv/azure/azure.go

```diff
@@ -0,0 +1,248 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+
+	"github.com/openclarity/vmclarity/testenv/types"
+	"github.com/openclarity/vmclarity/testenv/utils"
+	"github.com/openclarity/vmclarity/testenv/utils/docker"
+)
+
+const (
+	defaultRemoteUser                          = "vmclarity"
+	bicepFileResourcesPrefix                   = "vmclarity-"
+	bicepFileDiscoverRoleDeploymentNamePostfix = "-discover-role"
+	vmClarityServerPublicIPName                = bicepFileResourcesPrefix + "server-public-ip"
```

Comment:
lets use format definitions instead. also can be moved to files where these values are used (if not here then in azure utils)

```go
bicepFileResourcesFormat = "vmclarity-%s"
bicepFileDiscoverRoleDeploymentNameFormat = "%s-discover-role"
vmClarityServerPublicIPName = "vmclarity-server-public-ip"
```